File: PhaseJumpPro/Assets/phasejumppro/Model/Time/TimeSlice.cs
using System.Collections;

/*
 * RATING: 5 stars. Simple class
 * CODE REVIEW: 3.5.22
 */
namespace PJ {
	/// <summary>
	/// Encapsulates time delta for update events
	/// Allows us to add additional metadata for time events if needed
	/// </summary>
	public struct TimeSlice
	{
		public float delta;

		public TimeSlice(float delta) {
			this.delta = delta;
		}
	}
}

File: Swift/PhaseJumpPro/Sources/PhaseJumpPro/Time/TimeSlice.swift
import Foundation

/*
 RATING: 5 stars
 Simple type
 CODE REVIEW: 7/6/25
 */

/// Encapsulates time delta for update events
/// Allows for future time delta logic vs a standard float
public struct TimeSlice: Sendable
{
    public let delta: Double

    public init(delta: Double) {
        self.delta = delta
    }
    
    public static func *(lhs: Self, rhs: Self) -> Self {
        .init(delta: lhs.delta * rhs.delta)
    }
}
